src/js/popup/popup.js

```diff
@@ -1,802 +1,1350 @@
+              reportData.issue_on_domain.startsWith("http://") ||
+              reportData.issue_on_domain.startsWith("https://")
+            )
+          ) {
+            reportData.issue_on_domain = "http://" + reportData.issue_on_domain;
+          }
+
+          await browser.runtime.sendMessage({
+            method: "postReportWebcompatIssue",
+            description: reportData,
+          });
+        },
+      },
+    },
+    utilities: {
+      checkIfAnyMasksWereGeneratedOnCurrentWebsite: (masks, domain) => {
+        return masks.some((mask) => {
+          return domain === mask.generated_for;
+        });
+      },
+      clearBrowserActionBadge: async () => {
+        const { browserActionBadgesClicked } = await browser.storage.local.get(
+          "browserActionBadgesClicked"
+        );
+
+        // Dismiss the browserActionBadge only when it exists
+        if (browserActionBadgesClicked === false) {
+          browser.storage.local.set({ browserActionBadgesClicked: true });
+          browser.browserAction.setBadgeBackgroundColor({ color: null });
+          browser.browserAction.setBadgeText({ text: "" });
+        }
+      },
+      enableInputIconDisabling: async () => {
+        const inputIconVisibilityToggle = document.querySelector(
+          ".toggle-icon-in-page-visibility"
+        );
+
+        const stylePrefToggle = (inputsEnabled) => {
+          if (inputsEnabled === "show-input-icons") {
+            inputIconVisibilityToggle.dataset.iconVisibilityOption =
+              "disable-input-icon";
+            inputIconVisibilityToggle.classList.remove("input-icons-disabled");
+            return;
+          }
+          inputIconVisibilityToggle.dataset.iconVisibilityOption =
+            "enable-input-icon";
+          inputIconVisibilityToggle.classList.add("input-icons-disabled");
+        };
+
+        const iconsAreEnabled = await areInputIconsEnabled();
+        const userIconChoice = iconsAreEnabled
+          ? "show-input-icons"
+          : "hide-input-icons";
+        stylePrefToggle(userIconChoice);
+
+        inputIconVisibilityToggle.addEventListener("click", async () => {
+          const userIconPreference =
+            inputIconVisibilityToggle.dataset.iconVisibilityOption ===
+            "disable-input-icon"
+              ? "hide-input-icons"
+              : "show-input-icons";
+          await browser.runtime.sendMessage({
+            method: "updateInputIconPref",
+            iconPref: userIconPreference,
+          });
+          sendRelayEvent("Panel", "click", userIconPreference);
+          return stylePrefToggle(userIconPreference);
+        });
+      },
+      hasMaskBeenUsedOnCurrentSite: (mask, domain) => {
+        const domainList = mask.used_on;
+
+        // Short circuit out if there's no used_on entry
+        if (
+          domainList === null ||
+          domainList === "" ||
+          domainList === undefined
+        ) {
+          return false;
+        }
```

Comment:
Suggestion: Would this work too?
```suggestion
        if ([undefined, null, ""].includes(domainList)) {
          return false;
        }
```

Reply:
The first code snippet is checking if domainList is null, empty string, or undefined using the logical OR operator. If domainList matches any of these conditions, it will return false.

The second code snippet is using the Array.includes() method to check if domainList is included in the array [undefined, null, ""]. If domainList matches any of these values, it will return false.

So both code snippets are checking if domainList is a falsy value (null, empty string, or undefined) and returning false if it is.

Reply:
Thanks for the circle back! Not sure how I was reading this before. Revised in 6184ed0c24f894b4f7548b75076763f5d8518359